starlark/interp.go

```diff
@@ -83,8 +83,20 @@ func (fn *Function) CallInternal(thread *Thread, args Tuple, kwargs []Tuple) (Va
 
 	sp := 0
 	var pc uint32
-	var result Value
 	code := f.Code
+
+	defer func() {
+		if e := recover(); e != nil {
```

Comment:
Using recover is not part of the solution. If the interpreter encounters a panic, it's because the application code called by it has a bug, and the interpreter should not attempt to hide it.

There's still a question of whether the interpreter should use 'defer' in Call and CallInternal so that it pops the various stacks (Thread.stack, iterstack, profiler spans) to allow a panic to safely pass through the interpreter and be recovered in the application, without leaving the Thread in a bad state. It's an easy change, but whether we should make it depends on the cost it would add to returning from a Starlark function call in the normal case, for which we'd need a benchmark result.

Reply:
See #422